integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/massindexing/IndexWorkExecutorIT.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.backend.tck.massindexing;
+
+import static org.hibernate.search.util.impl.integrationtest.common.assertion.SearchResultAssert.assertThat;
+import static org.hibernate.search.util.impl.integrationtest.common.stub.mapper.StubMapperUtils.referenceProvider;
+
+import java.time.LocalDate;
+import java.time.Month;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+import org.hibernate.search.engine.backend.document.DocumentElement;
+import org.hibernate.search.engine.backend.document.IndexFieldAccessor;
+import org.hibernate.search.engine.backend.document.model.dsl.IndexSchemaElement;
+import org.hibernate.search.engine.backend.index.spi.IndexWorkExecutor;
+import org.hibernate.search.engine.search.DocumentReference;
+import org.hibernate.search.engine.search.SearchPredicate;
+import org.hibernate.search.engine.search.SearchQuery;
+import org.hibernate.search.engine.search.dsl.predicate.SearchPredicateFactoryContext;
+import org.hibernate.search.integrationtest.backend.tck.util.rule.SearchSetupHelper;
+import org.hibernate.search.util.impl.integrationtest.common.stub.mapper.StubMappingIndexManager;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+/**
+ * Verify that the {@link IndexWorkExecutor}, provided by a backend, is working properly, storing correctly the indexes.
+ */
+public class IndexWorkExecutorIT {
+
+	private static final String INDEX_NAME = "bestSellingBooksEver";
+
+	@Rule
+	public SearchSetupHelper setupHelper = new SearchSetupHelper();
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	private IndexAccessors indexAccessors;
+	private StubMappingIndexManager indexManager;
+
+	@Before
+	public void before() {
+		setupHelper.withDefaultConfiguration()
+				.withIndex(
+						"MappedType", INDEX_NAME,
+						ctx -> this.indexAccessors = new IndexAccessors( ctx.getSchemaElement() ),
+						indexManager -> this.indexManager = indexManager
+				)
+				.setup();
+
+		initData();
+	}
+
+	@Test
+	public void findBulk() {
+		SearchQuery<DocumentReference> query = queryWithPredicate( f -> f.matchAll().toPredicate() );
```

Comment:
Not sure we need anything more than this test, really. Checking that the index contains everything that you wanted to index should be enough. The various field types, predicates, etc. are already tested elsewhere.

More importantly, the point of the work executor is to index *lots* of documents. Here, by complicating your schema, you forced yourself to write the value for each document by hand, and thus you couldn't index more than a few documents.

I think you should use a dramatically simpler schema (just one String field) and index a few hundred documents instead of just five. Not that I expect anything to go wrong: it's just what this interface is for, so let's test it like it is supposed to be used.